src/libraries/Microsoft.PowerFx.Core/Functions/UserDefinedFunction.cs

```diff
@@ -79,7 +79,7 @@ public void CheckTypesOnDeclaration(CheckTypesContext context, DType actualBodyR
 
             var returnTypeFormulaType = _returnTypeToken.GetFormulaType()._type;
 
-            if (!returnTypeFormulaType.Kind.Equals(actualBodyReturnType.Kind) || !returnTypeFormulaType.CoercesTo(returnTypeFormulaType))
+            if (!returnTypeFormulaType.Kind.Equals(actualBodyReturnType.Kind) || !returnTypeFormulaType.CoercesTo(returnTypeFormulaType, true, true, context.Features.PowerFxV1CompatibilityRules))
```

Comment:
>true [](http://example.com/codeflow?start=143&length=4)

This probably should be false as default value for isTopLevelCoercion is false. Also use named param syntaxso it's easy to read.